trin-core/src/utp/stream.rs

```diff
@@ -395,11 +84,19 @@ impl UtpListener {
             Ok(packet) => {
                 let connection_id = packet.connection_id();
 
+                // Only handle packets if they are on our watchlist and have been negotiated
+                // I believe this is what is meant in the specs when they say after getting this message
+                // Listen for a response
+                // [discv5-utp]: https://github.com/ethereum/portal-network-specs/blob/master/discv5-utp.md?plain=1#L26
+
+                // todo: uncomment this at a future date, as this makes it harder to test with other
+                // clients for the time being
```

Comment:
It would be helpful to include the trigger here, instead of just "later". Does this mean: the other clients have implemented utp differently, so we can switch when the change the implementation somehow? What needs to change in those clients? How can we confirm/test that that has happened? Otherwise, this will never change.

Really, my intuition here is to not put code in comments, and instead open a second WIP PR with these changes, and a description that answers the above questions.